pkg/skaffold/deploy/kubectl/cli.go

```diff
@@ -73,6 +75,50 @@ func (c *CLI) Apply(ctx context.Context, out io.Writer, manifests ManifestList)
 	return nil
 }
 
+type getResult struct {
+	Items []resource `json:"items"`
+}
+
+type resource struct {
+	Metadata metadata `json:"metadata"`
+}
+
+type metadata struct {
+	Name              string `json:"name"`
+	DeletionTimestamp string `json:"deletionTimestamp"`
+}
+
+// GetPendingDeletions uses `kubectl get` to list resources that are marked for deletion.
+func (c *CLI) GetPendingDeletions(ctx context.Context, manifests ManifestList) ([]string, error) {
+	args := []string{"-f", "-", "--ignore-not-found=true", "-ojson"}
```

Comment:
I wonder if kubectl can be tweaked to do the filtering...

Reply:
Only possible with `-ojsonPath` but then it doesn't output json anymore...